packages/fe-fpm-writer/src/building-block/index.ts

```diff
@@ -0,0 +1,155 @@
+import { render } from 'ejs';
+import type { Editor } from 'mem-fs-editor';
+import { join } from 'path';
+import type { BuildingBlock } from './types';
+import { DOMParser, XMLSerializer } from '@xmldom/xmldom';
+import * as xpath from 'xpath';
+import format from 'xml-formatter';
+import { getErrorMessage, validateBasePath } from '../common/validate';
+
+/**
+ * Generates a building block into the provided xml view file.
+ *
+ * @param {string} basePath - the base path
+ * @param {string} viewPath - the path of the xml view relative to the base path
+ * @param {string} aggregationPath - the aggregation xpath
+ * @param {BuildingBlock} buildingBlockData - the building block data
+ * @param {Editor} [fs] - the memfs editor instance
+ * @returns {Editor} the updated memfs editor instance
+ */
+export function generateBuildingBlock<T extends BuildingBlock>(
+    basePath: string,
+    viewPath: string,
+    aggregationPath: string,
+    buildingBlockData: T,
+    fs: Editor
+): Editor {
+    // Validate the base and view paths
+    validateBasePath(basePath, fs);
+    if (!fs.exists(join(basePath, viewPath))) {
+        throw new Error(`Invalid view path ${viewPath}.`);
+    }
+
+    // Read the view xml and template files and update contents of the view xml file
+    const viewDocument = getViewDocument(basePath, viewPath, fs);
```

Comment:
Can we also handle fragments?

Reply:
The code should work for both xml views and xml fragment files. I updated the public function's variable name and docs to indicate this. I haven't updated this in other places in the file for better readability.